### PR TITLE
Add versioning

### PR DESCRIPTION
**Pull request by swinkels**
Hi Charles, 

This pull request adds versioning to git-cc. It is not yet ready for merge but I wanted to have your feedback first. I noticed :) you are not a proponent of explicit version numbers in the code and that you would prefer getting the version from the tag. However that is not the way it is implemented in the pull request...

As mentioned in an earlier comment, the versioning I would like to add makes use of [zest.releaser](https://github.com/zestsoftware/zest.releaser), a Python package to automate releases. If you want to have an idea of how it works, I added a section in the README to describe it. Of course you can always try it out.

Please let me know what you think of it.

### CODE REVIEW

**Comment by charleso** [comment]
@swinkels As someone who doesn't use python _or_ git-cc (any more) I'm not sure how important or useful my opinion is these days. If this is the best way to package up git-cc it's fine by me.

When you're done, and have rebased on `master` let me know and I'll take another look and then merge.

File: setup.py [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+
 from setuptools import setup, find_packages
 
+# The following approach to retrieve the version number is inspired by this
+# comment:
+#
+#   https://github.com/zestsoftware/zest.releaser/issues/37#issuecomment-14496733
+#
+# With this approach, development installs always show the right version number
+# and do not require a reinstall (as the definition of the verdion umber in
```
Comment (charleso) [inline setup.py]: `s/verdion umber/version number/`

**Comment by swinkels** [comment]
You are the primary author of the tool and as such, I consider you to be 'upstream'. As is proper for an open source tool, I would like to push my changes upstream. However, I can imagine that if you do not use git-cc anymore, or even Python, that reviewing those changes and merging them, takes more time than it is worth. Just let me know and I will leave future changes in my fork. People that are interested can pick them up from there.

Note that I am not talking about taking over maintainership. In my current job I have to use ClearCase and I would like it to become a bit more useful for my colleagues and me. So with my changes I am just scratching a personal itch here :)

**Comment by charleso** [comment]
> You are the primary author of the tool and as such, I consider you to be 'upstream'. As is proper for an open source tool, I would like to push my changes upstream.

Sorry if I sounded irritated, it doesn't bother me at all reviewing the PR, was just making a statement about how just how irrelevant my opinion in on anything related to Python. :)

> Just let me know and I will leave future changes in my fork. People that are interested can pick them up from there.

Despite what I said above, I _think_ it's still better to keep pulling what I can to this top-level repository unless someone really wants to take it over and I'm happy to put a notice in the README letting people know where to look.

Thanks for all the interest, and I hope at the end of the day git-cc is vaguely useful (and apologies for the state of the code, I was young and stupid when I wrote it).

**Comment by swinkels** [comment]
> Sorry if I sounded irritated, it doesn't bother me at all reviewing the PR, was just making a 
> statement about how just how irrelevant my opinion in on anything related to Python

Don't worry, you did not sound irritated.

> and I hope at the end of the day git-cc is vaguely useful (and apologies for the state of the code, I 
> was young and stupid when I wrote it).

It is extremely useful. For my current job I have to use dynamic ClearCase views. Coming from Git, I feel like ClearCase is only in the way. Your Git bridge is one of the pieces of the puzzle to become more productive. So thank you for that!

git-cc does have some rough edges and those I would like to address. :) To do so, I would like to have proper versioning, tests, better documentation, etc. and the first PRs will address that.

**Comment by swinkels** [comment]
All required changes are in, so this pull request is ready for merge.

I would like to propose to make a release after this pull request is merged. That released version will be version 1.0.0 and then we can start working on version 1.0.1dev0. Creating a release is just one command, as described in the README. 

However, if for any reason you cannot make the release, I can do the 1.0.0 release on my repo. The changes caused by the release are really, really minor. If you then pull in my master branch, maybe via a pull request, you also have the release.

**Comment by charleso** [comment]
@swinkels I'm feeling quite lazy, if I added you as a contributor would you be ok to run it against this repository?

**Comment by charleso** [comment]
@swinkels Scrap that, have released as you suggested.

Thanks for everything. :)

**Comment by swinkels** [comment]
Thanks to you!

**Pull request merged**